include/devicetree.h

```diff
@@ -768,6 +768,32 @@
 #define DT_STRING_TOKEN(node_id, prop) \
 	DT_CAT4(node_id, _P_, prop, _STRING_TOKEN)
 
+/**
+ * @brief Like DT_STRING_TOKEN(), but with a fallback to default_value
+ *
+ * If the value exists, this expands to DT_STRING_TOKEN(node_id, prop).
+ * The default_value parameter is not expanded in this case.
+ *
+ * Otherwise, this expands to default_value.
+ *
+ * @param node_id node identifier
+ * @param prop lowercase-and-underscores property name
+ * @param default_value a fallback value to expand to
+ * @return the property's value or default_value
+ */
+#define DT_STRING_TOKEN_OR(node_id, prop, default_value) \
```

Comment:
Nit: the name `DT_STRING_TOKEN_OR_DEFAULT` is a little more explicit.

I initially read it like it was a logical or operation and had to go figure out what it was doing.